reacthomeworkweek2/App.js

```diff
@@ -0,0 +1,90 @@
+'use strict';
+
+class Container extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      text: { description: '', deadline: '', done: false },
+      items: [
+        {
+          description: 'Get out of bed',
+          deadline: '2017-09-11',
+          done: true,
+        },
+        {
+          description: 'Brush teeth',
+          deadline: '2017-09-10',
+          done: false,
+        },
+        {
+          description: 'Eat breakfast',
+          deadline: '2017-09-09',
+          done: false,
+        },
+      ],
+    };
+  }
+  handleDelete(index) {
+    event.preventDefault();
+    const items = [...this.state.items];
+
+    items.splice(index, 1);
+    this.setState({ items });
+  }
+  handleSubmit(event) {
+    event.preventDefault();
+    let text = { ...this.state.text };
+    const items = [...this.state.items];
+    text.description && text.deadline ? items.push(text) && this.setState({ items }) : null;
+  }
+  handleChange(event) {
+    const text = { ...this.state.text };
+    const { name, value } = event.target;
+    text[name] = value;
+    this.setState({ text });
+  }
+
+  render() {
+    return (
+      <div>
+        <p id="todo">Todo</p>
+        <form onSubmit={this.handleSubmit.bind(this)}>
+          <label id="title">Description:</label>
+          <input
+            id="input"
+            type="text"
+            onChange={this.handleChange.bind(this)}
+            name="description"
+          />
+          <label id="title">Deadline:</label>
+          <input id="input" type="date" onChange={this.handleChange.bind(this)} name="deadline" />
+          <button id="add">Add</button>
+        </form>
+
+        {this.state.items.map((item, index) => (
+          <div key={index}>
```

Comment:
If you're having an ids using them would a valuable thing
using indexes can cause in bugs.

using indexes is might replaces the ids but only in case of not having them as the [docs](https://reactjs.org/docs/lists-and-keys.html) says:
> When you don’t have stable IDs for rendered items, you may use the item index as a key

And beyond that they're saying:
> We don’t recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state.